### DISCUSSION TITLE
OpenWRT API?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [pulumi/pulumi#13907] [General]
Does anyone know of pulumi API for openwrt.org? Trying to automate my home lab and need a way to configure my home router

**Comment by pcuci** [comment]
Yeah, it's a mess :-) Haven't found anything yet, looking into https://www.pulumi.com/docs/concepts/resources/dynamic-providers/ to perhaps roll up a simple one on my own?

My specific use-case is creating Proxmox VMs with DHCP'd IPs. For some odd reason my OpenWRT-based [Turris](https://www.turris.com/) router decides to hand over the same dynamic IP to different MAC addresses(?), so I was thinking I could instead tell an "OpenWrtDhcpRouter dynamic provider" to reserve a 50 addresses block just for ProxmoxVE, and programmatically refresh state against it on every `pulumi up` via: https://github.com/pulumi/pulumi/issues/8058

How about using the [Pulumi Terraform Bridge](https://github.com/pulumi/pulumi-terraform-bridge) for this one, @hrstoyanov: https://registry.terraform.io/providers/joneshf/openwrt/latest/docs ?
